internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -862,8 +864,42 @@ func (r *BMACReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		}}}
 	}
 
+	mapInfraEnvToBMH := func(a client.Object) []reconcile.Request {
+		ctx := context.Background()
+		infraEnv := &aiv1beta1.InfraEnv{}
+
+		if err := r.Get(ctx, types.NamespacedName{Name: a.GetName(), Namespace: a.GetNamespace()}, infraEnv); err != nil {
+			return []reconcile.Request{}
+		}
+
+		// Don't queue any reconcile if the InfraEnv
+		// doesn't have the ISODownloadURL set yet.
+		if infraEnv.Status.ISODownloadURL == "" {
+			return []reconcile.Request{}
+		}
+
+		bmhs, err := r.findBMHByInfraEnv(ctx, infraEnv)
+		if len(bmhs) == 0 || err != nil {
+			return []reconcile.Request{}
+		}
+
+		requests := []reconcile.Request{}
```

Comment:
Minor tweak: since you know the length, you can go ahead and `make` this the correct size, allocating the final structure all at once. Then you can just assign values to each location in the slice instead of using `append`.